framework/src/base/EigenProblem.C

```diff
@@ -124,6 +124,15 @@ EigenProblem::computeJacobian(const NumericVector<Number> & soln, SparseMatrix<N
   FEProblemBase::computeJacobian(soln, jacobian, kernel_type);
 }
 
+void
+EigenProblem::checkProblemIntegrity()
+{
+  FEProblemBase::checkProblemIntegrity();
+#if LIBMESH_HAVE_SLEPC
```

Comment:
If there does not exist SLEPc,  `NonlinearEigenSystem` is a shell class (the class has the constructor only). ` _nl_eigen ` is a pointer of the instance of `NonlinearEigenSystem`. ` _nl_eigen` has no function member `checkIntegrity()` anymore if SLEPc is not available.